lfs.h

```diff
@@ -406,6 +420,8 @@ typedef struct lfs {
     lfs_size_t name_max;
     lfs_size_t file_max;
     lfs_size_t attr_max;
+    lfs_size_t metadata_max;
```

Comment:
Ah, if `metadata_max` is not dependent on the superblock, it's better to not keep it in the `lfs_t` struct, but instead calculate its value on every use (`lfs->metadata_max ? lfs->metadata_max : lfs->cfg->block_size`).

Something similar is done for `block_cycles`. It's unfortunately a bit noisy, but it reduces RAM at the cost of code size, and has the potential to be optimized out in https://github.com/littlefs-project/littlefs/pull/491.

Reply:
Ah, I was a little worried at how sparse `lfs_t` was and whether this was the right choice. I'll modify this to calculate-at-use when I sort the inline files config.